Fabric/src/main/java/com/techarium/techarium/platform/FabricPlatformHelper.java

```diff
@@ -0,0 +1,127 @@
+package com.techarium.techarium.platform;
+
+import com.techarium.techarium.platform.services.IPlatformHelper;
+import net.fabricmc.fabric.api.client.render.fluid.v1.FluidRenderHandlerRegistry;
+import net.fabricmc.fabric.api.screenhandler.v1.ExtendedScreenHandlerFactory;
+import net.fabricmc.fabric.api.transfer.v1.context.ContainerItemContext;
+import net.fabricmc.fabric.api.transfer.v1.fluid.FluidConstants;
+import net.fabricmc.fabric.api.transfer.v1.fluid.FluidStorage;
+import net.fabricmc.fabric.api.transfer.v1.fluid.FluidVariant;
+import net.fabricmc.fabric.api.transfer.v1.fluid.FluidVariantAttributes;
+import net.fabricmc.fabric.api.transfer.v1.storage.Storage;
+import net.fabricmc.fabric.api.transfer.v1.storage.StorageView;
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.MenuProvider;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.inventory.InventoryMenu;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.material.Fluid;
+import net.minecraft.world.level.material.Fluids;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.Consumer;
+
+public class FabricPlatformHelper implements IPlatformHelper {
+
+	@Override
+	public String getPlatformName() {
+		return "Fabric";
+	}
+
+	@Override
+	public boolean isModLoaded(String modId) {
+
+		return FabricLoader.getInstance().isModLoaded(modId);
+	}
+
+	@Override
+	public boolean isDevelopmentEnvironment() {
+
+		return FabricLoader.getInstance().isDevelopmentEnvironment();
+	}
+
+	@Override
+	public void openGui(ServerPlayer player, MenuProvider provider, Consumer<FriendlyByteBuf> extraData) {
+		ExtendedScreenHandlerFactory factory = new ExtendedScreenHandlerFactory() {
```

Comment:
Cleaner and easier to read/maintain.
Also, with this change, I would suggest letting the screen handle reading the buffer through the constructor rather than in your registry implementations as it will allow you to send more data when it gets opened without having to send other packets or such.